### DISCUSSION TITLE
Debugging smart plugs (Avatar AWP07L) that randomly reset?

### DISCUSSION THREAD

**Discussion by shawly** [arendst/Tasmota#21520] [Support request]
I have around sixteen Avatar AWP07L smart plugs at home that I bought back in 2020.

I flashed them with Tuya Convert to Tasmota and calibrated them so I can keep track of my power consumption.

Now quite a few of them are seemingly starting to fail and randomly reboot, I have three that I never used but recently updated and now they reboot constantly after a few seconds. The most annoying issue is, that they also turn off the relay on reboots. This is currently affecting the smart plug which my router is powered by, so when that plug randomly reboots (Reboot Reason is usually "Power On" or "Software/System restart", my Wi-fi drops out and then many other smart plugs also seem to start rebooting (probably because of the wifi loss?). Strangely enough some of the smart plugs never rebooted since 152 days and are just fine even when wifi drops for a few minutes.

I guess this is the fate of cheap hardware, but I don't like throwing electronics away before I haven't had at least one attempt at repairing.

I guess I'll have to solder on some wires to TX and RX so I can actually debug the issue, my first guess is that the ones that randomly reset are browning out maybe?

So what should be the first thing to check for? Cold solder joints? Broken capacitors? Fried power circuit? Not sure which components would actually be faulty here.

**Comment by sfromis** [comment]
While I have no experience with that plug, there is a clear precedent for cheap hardware barely able to power itself, and getting worse over time. Might be capacitors, or something else.....

In some cases, simpler/older Tasmota builds may tax the power generation a bit less, and be more stable. Still, it is "pretty plausible" that such plugs are up for either repair or electronics recycling....
